RefApp/ViewControllers/AssetListTableViewController.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  AssetListTableViewController.swift
+//  RefApp
+//
+//  Created by Udaya Sri Senarathne on 2018-11-21.
+//  Copyright © 2018 emp. All rights reserved.
+//
+
+import UIKit
+import Exposure
+import ExposurePlayback
+
+class AssetListTableViewController: UITableViewController {
+    
+    var assets = [Asset]()
+    var sessionToken: SessionToken?
+    
+    var datasource:TableViewDataSource<Asset>?
+    let cellId = "assetListTableViewCell"
+    
+    // MARK: Life Cycle
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        self.navigationController?.setNavigationBarHidden(false, animated: false)
+        self.assetsDidLoad([])
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.navigationItem.title = NSLocalizedString("Assets", comment: "")
+        self.navigationController?.navigationBar.prefersLargeTitles = true
+        self.navigationController?.navigationBar.barStyle = .black
+        
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: cellId)
+        tableView.tableFooterView = UIView()
+        
+        addLogoutBarButtonItem()
+        self.generateTableViewContent()
+    }
+    
+    
+    /// Add left bar button item
+    fileprivate func addLogoutBarButtonItem() {
+        let button = UIButton()
+        button.addTarget(self, action:#selector(handleLogout), for: .touchUpInside)
+        button.setTitle(NSLocalizedString("Logout", comment: ""), for: .normal)
+        button.setTitleColor(UIColor.white, for: .normal)
+        let barButton = UIBarButtonItem(customView: button)
+        self.navigationItem.leftBarButtonItem = barButton
+    }
+}
+
+// MARK: - DataSource
+extension AssetListTableViewController {
+    
+    /// Generate tableview content by loading assets from API
+    fileprivate func generateTableViewContent() {
+        guard let environment = StorageProvider.storedEnvironment, let _ = StorageProvider.storedSessionToken else {
```

Comment:
A session token is not required to fetch assets but the guard statement indicates this is the case.

I would prefer checking the sessionToken once it is required, ie when making a play call. If no token is available then, prompt the user to login before making the play request

Reply:
I thought it is good to check the session token before fetching assets. But may be as it is not needed to fetch the assets, there is no point of checking for the token. Will remove it from the guard statement.